### DISCUSSION TITLE
What is the process to get YMAL author contribution information to actually show up in the rendered document?

### DISCUSSION THREAD

**Discussion by alexfradera** [quarto-dev/quarto-cli#9456] [Q&A] [answered]
### Description

As per the [documentation](https://quarto.org/docs/authoring/front-matter.html#authors-and-affiliations), it is relatively simple to introduce information about author roles/contribution as below. But how to get this generated into my rendered output? I can't seem to find anything in the documentation that acknowledges this; I have previously done this in rmarkdown using papaja so seeking to find some way to produce the output. 

Thanks very much for any support.

````qmd
---
title: "Reproducible Quarto Document"
format: docx
author:
  - name: Josiah Carberry
    roles: 
      - investigation: lead 
      - data curation: supporting
---
````

**Comment by mcanouil** [comment, marked as answer]
You need to define a [custom Word document template](https://quarto.org/docs/output-formats/ms-word-templates.html) (I am not sure it's really possible that way) or use a Lua filter such as https://github.com/kapsner/authors-block

Note that the documentation you shared is only the full schema. It does not mean that default formats/templates are using every pieces of it.

**Reply by alexfradera** [reply]
Thank you. I looked at the custom Word templates but like you didn't see a way to produce this. It seems that a Lua filter is the way to go, perhaps I will try and figure it out. Thanks again.

**Reply by Antoine-Creon** [reply]
Hi @alexfradera did you figure out a way to produce a Lua filter ? I'd like to generate information about author roles/contribution using quarto's keys too. Kapsner's authors-block is terrific but I'm not proficient enough with Pandoc/Lua to write such a filter myself. Thanks !

**Reply by mcanouil** [reply]
Note that `authors` is following a schema, see <https://quarto.org/docs/journals/authors.html#author-schema>.
You cannot use something that is not on, the schema for `authors`.

**Reply by Antoine-Creon** [reply]
Yes I read this page carefully. Maybe I am misunderstanding what your are saying but to me roles are part of the [schema for authors](https://quarto.org/docs/journals/authors.html#roles) and thus could be used ?

**Reply by mcanouil** [reply]
If that's part of the schema, then yes you can use it.

If you are looking for example, look at template and extensions, as the one I initially shared.
